### PR TITLE
dapp console: add mixpanel 

### PR DESCRIPTION
**Pull request by tarunkhasnavis**
closes https://github.com/ethereum-optimism/ecosystem/issues/103

This is the initial PR for Mixpanel setup for Dapp Console. You can change mixpanel project tokens based on environment type (we have two: testing and production).

### CODE REVIEW

**Comment by tarunkhasnavis** [comment]
Current dependencies on/for this PR:
* **#136** <a href="https://app.graphite.dev/github/pr/ethereum-optimism/ecosystem/136?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `main`

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/ethereum-optimism/ecosystem/136?utm_source=stack-comment).

File: apps/dapp-console/app/mixpanel/mixpanel.tsx [hunk 1]
```diff
@@ -0,0 +1,14 @@
+import mixpanelBrowser from 'mixpanel-browser'
+
+const initMixpanel = () => {
+  if (
```
Comment (nitaliano) [inline apps/dapp-console/app/mixpanel/mixpanel.tsx:4]: would just do `if(!process.env.VITE_MIXPANEL_TOKEN)` then we can have the system setup so if it doesn't exist we just don't init mixpanel, we can ignore passing in incorrect strings

File: apps/dapp-console/app/mixpanel/mixpanel.tsx [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+    process.env.VITE_MIXPANEL_TOKEN === 'false' ||
+    process.env.VITE_MIXPANEL_TOKEN == null
+  ) {
+    return null
+  }
+  mixpanelBrowser.init(process.env.VITE_MIXPANEL_TOKEN)
+  return mixpanelBrowser
+}
+
+export const mixpanel = initMixpanel()
```
Comment (nitaliano) [inline apps/dapp-console/app/mixpanel/mixpanel.tsx:14]: not sure how you're going to implement the tracking portion, but i think we should avoid using this directly in our component files like we do in gateway meaning this

```
mixpanel?.track(TRACKING_EVENT_NAME.PRETEND_EVENT, 
  [MIXPANEL_CUSTOM_TRACKING_PROPERTY.PRETEND]: 'pretend',
})
```

having to import the custom tracking properties especially if we ever want to use the same event track call in multiple places and having mixpanel across the codebase would make it harder to refactor if we ever dropped mixpanel

instead i think we should wrap all them with functions and keep them under this directory so we can have functions that do this

```
function trackPretendEvent(args: PretendEventArgs) {
    mixpanel?.track('pretendEvent', 
     'Pretend': args.pretendProperty,
   })
}
```

**Comment by nitaliano** [comment]
potentially also drop the `mixpanel` directory name for something more generic like `event-tracking` or something along those lines so the imports also stay generic

File: apps/dapp-console/app/mixpanel/mixpanel.tsx (hunk 2 above)
Comment (tarunkhasnavis) [reply apps/dapp-console/app/mixpanel/mixpanel.tsx:14]: oh cool, so the usage would be like 

```
import { trackPretendEvent } from 'event-tracking/mixpanel'

// on the action to track 
trackPretendEvent(relevantArgs)
```

and for each new event name there would be a new event function added to mixpanel.ts taking in the relevant arguments

Comment (nitaliano) [reply apps/dapp-console/app/mixpanel/mixpanel.tsx]: Yup exactly and if the file ever gets too long can be broken out into multiple

**Review by nitaliano — APPROVED** [review]

**Pull request merged**